src/strategies/backscratcher/strategy-frax-temple-univ2.sol

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "../strategy-base.sol";
+import "../../interfaces/backscratcher/IStrategyProxy.sol";
+import "../../interfaces/backscratcher/FraxGauge.sol";
+import "../../interfaces/templeRouter.sol";
+
+contract StrategyFraxTempleUniV2 is StrategyBase {
+    address public strategyProxy;
+
+    IERC20 public frax_temple_pool = IERC20(0x6021444f1706f15465bEe85463BCc7d7cC17Fc03);
+    address public frax_temple_gauge = 0x10460d02226d6ef7B2419aE150E6377BdbB7Ef16;
+
+    address public constant templeRouter = 0x8A5058100E60e8F7C42305eb505B12785bbA3BcA;
+
+    address public constant FXS = 0x3432B6A60D23Ca0dFCa7761B7ab56459D9C964D0;
+    address public constant TEMPLE = 0x470EBf5f030Ed85Fc1ed4C2d36B9DD02e77CF1b7;
+    address public constant FRAX = 0x853d955aCEf822Db058eb8505911ED77F175b99e;
+
+    address[] public rewardTokens = [FXS, TEMPLE];
+
+    constructor(
+        address _governance,
+        address _strategist,
+        address _controller,
+        address _timelock
+    ) public StrategyBase(address(frax_temple_pool), _governance, _strategist, _controller, _timelock) {}
+
+    // **** Views ****
+    function setStrategyProxy(address _proxy) external {
+        require(msg.sender == governance || msg.sender == strategist, "!governance");
+        strategyProxy = _proxy;
+    }
+
+    function getName() external pure override returns (string memory) {
+        return "StrategyFraxTempleUniV2";
+    }
+
+    // **** State Mutations ****
+
+    function harvest() public override onlyBenevolent {
+        IStrategyProxy(strategyProxy).harvest(frax_temple_gauge, rewardTokens);
+
+        uint256 _fxs = IERC20(FXS).balanceOf(address(this));
+        IERC20(FXS).safeApprove(univ2Router2, 0);
+        IERC20(FXS).safeApprove(univ2Router2, _fxs);
+
+        address[] memory _path = new address[](2);
+        _path[0] = FXS;
+        _path[1] = FRAX;
+        _swapUniswapWithPath(_path, _fxs);
+
+        uint256 _frax = IERC20(FRAX).balanceOf(address(this));
+
+        IERC20(FRAX).safeApprove(templeRouter, 0);
```

Comment:
I lean towards giving infinite approvals in the constructor, but that's a matter of preference.

Reply:
Agreed, would save us on a bit of gas too.